### PR TITLE
[AI] AI Rework 2.5 || The one with cables and golem AIs

### PR DESCRIPTION
**Pull request by Bibby0110**
# Document the changes in your pull request

Instead of AI data cores always being connected you'll now have to wire them up using ethernet cables.
The "home" network of the AI is the one where its current core is. If the current core fails it will connect to a different core in the network. 
You can link 2 remote networks with networking devices (sprites pending, currently solar panels in the screenshot). In this case the AI will also attempt to relocate to a remote core, and switch it's home network. This is done recursively so you can nest networks if you wanted that for some reason.
If you have a data core in a separate network, aka in a network that isn't the AIs main network, you'll need a server cabinet with 2Thz of CPU and 3TB of RAM for the core to be "active"

Explanatory picture (Not actual roundstart setup):
image
Network outlined in red contains the core and a server cabinet. Also contains a networking device(solar panel)
Network outlines in blue contains only a server cabinet and networking device.
The two networking devices are connected, connecting the red and blue networks and allowing them to share processing power. Connection signified by black line. (Note that solar panels point towards each other, allowing for simple tracking of clusters)
### Why?
Why this change you may ask? Mainly for the following 3 reasons (There are more that I have forgotten):

1. Removal of the AI control console. 
Making AIs confined to physical networks instead of *all* data cores means it's now possible to model hacking into the AIs network physically, without being able to build a new core and doing it there. This way you'll need physical access to download the AI like before.
2. Separation of networks
Wanna run 2 AIs with separate resources? You can do that now.
Wanna build an AI on lavaland? You can do that now (Reenables AIs for ghost roles)
3. Making computing networks independent
Having computer networks be physical like this allows us to use them for things that aren't just AI related. This way the Netmin will be able to use the processing clusters for bitcoin mining, research and [REDACTED, COMING SOON]

### TODO
The things I am yet to complete

- [ ] Require local computing resources for data cores to be active
- [x] Actually consider local resources for AIs
- [ ] Resource allocation console moved
- [ ] Remove the control console
- [ ] UI for networking devices
- [ ] Allow plugging into individual AI networks
- [ ] Allow uploading of AIs after plugging in
- [ ] Allow downloading of AIs after plugging in
- [ ] Allow hacking and overwriting of networking machines?
- [ ] Allow individual control of networks through laptops. Allow for bitcoin mining on networks with no AI/cores present
- [ ] A sprite for networking machines maybe
- [ ] Map in networking machines on each map
- [ ] (Expand box sci-maint to create a proper netmin area?)
- [ ] (Decouple netmin from NTSL?)

# Changelog

### CODE REVIEW

**Comment by ShadowDeath6** [comment]
How would you go about making an AI after this? Protolathe a modular computer + shove a posibrain in there after you make the requisite machinery?

**Comment by Bibby0110** [comment]
> How would you go about making an AI after this? Protolathe a modular computer + shove a posibrain in there after you make the requisite machinery?

Connect a laptop to the network and upload the MMI or posibrain

**Comment by Xoxeyos** [comment]
Can you please add a way to revive dead AIs, as well as a sprite for the cores that may display overheating?

File: yogstation/code/modules/jobs/job_types/network_admin.dm [hunk 1]
```diff
@@ -15,7 +15,7 @@
 	outfit = /datum/outfit/job/network_admin
 
 	added_access = list(ACCESS_ENGINE, ACCESS_ENGINE_EQUIP, ACCESS_MAINT_TUNNELS)
-	base_access = list(ACCESS_TCOMSAT, ACCESS_TCOM_ADMIN, ACCESS_TECH_STORAGE, ACCESS_RC_ANNOUNCE, ACCESS_CONSTRUCTION, ACCESS_MECH_ENGINE, ACCESS_NETWORK, ACCESS_RESEARCH, ACCESS_MINISAT)
+	base_access = list(ACCESS_TCOM_ADMIN, ACCESS_TECH_STORAGE, ACCESS_RC_ANNOUNCE, ACCESS_CONSTRUCTION, ACCESS_MECH_ENGINE, ACCESS_NETWORK, ACCESS_RESEARCH, ACCESS_MINISAT, ACCESS_TOX)
```
Comment (redmoogle) [inline yogstation/code/modules/jobs/job_types/network_admin.dm:18]: Fix

**Review by redmoogle — CHANGES_REQUESTED** [review]

File: yogstation/code/modules/jobs/job_types/network_admin.dm (hunk 1 above)
Comment (Bibby0110) [reply yogstation/code/modules/jobs/job_types/network_admin.dm:18]: not a bug?

Comment (adamsong) [reply yogstation/code/modules/jobs/job_types/network_admin.dm:18]: Give them back ACCESS_TCOMSAT

**Review by adamsong — CHANGES_REQUESTED** [review]

File: yogstation/code/modules/jobs/job_types/network_admin.dm (hunk 1 above)
Comment (Bibby0110) [reply yogstation/code/modules/jobs/job_types/network_admin.dm:18]: nope

Comment (adamsong) [inline yogstation/code/modules/jobs/job_types/network_admin.dm:18]:
```suggestion
	base_access = list(ACCESS_TCOMSAT, ACCESS_TCOM_ADMIN, ACCESS_TECH_STORAGE, ACCESS_RC_ANNOUNCE, ACCESS_CONSTRUCTION, ACCESS_MECH_ENGINE, ACCESS_NETWORK, ACCESS_RESEARCH, ACCESS_MINISAT, ACCESS_TOX)
```

**Comment by adamsong** [comment]
I liked this pr, except for where you removed a job people like

**Comment by Bibby0110** [comment]
> I liked this pr, except for where you removed a job people like

good 4 u

**Pull request closed without merging**